README.md

```diff
@@ -288,13 +288,13 @@ brew upgrade icu4c
 To compile for macOS X on x86:
 
 ```bash
-go build --tags "darwin,amd64"
```

Comment:
These instructions were actually totally wrong. It should be using `GOOS` and `GOARCH` for this. The same applies for Windows and Linux. (Also the flag is `-tags` not `--tags`.)

Reply:
Sorry, I'm not following where's my fault here. I've added architecture flags exactly in a way it was for OS selection: by explicitly define target OS/arch. I'm a roocky in cgo and not sure how to re-write this to make it refer to `GOOS` and `GOARCH` env vars: `#cgo` flag does not have `if-then-else` notion.

Reply:
The original instructions (on master) were incorrect.

If you are doing a _native_ compile, it is just `go build`.
If you are doing a _cross_ compile, it is `GOOS=<target-os> GOARCH=<target-arch> CGO_ENABLED=1 CC=<c-cross-compiler> go build`
For example: `GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 CC=o64-clang go build`

Separately from that, if you are trying to include optional features via build tags, in modern versions of Go you provide a comma-separated list via `-tags`.
> -tags tag,list
	a comma-separated list of build tags to consider satisfied during the
	build. For more information about build tags, see the description of
	build constraints in the documentation for the go/build package.
	(Earlier versions of Go used a space-separated list, and that form
	is deprecated but still recognized.)

For example: `go build -tags sqlite_stat4,secure_delete` 

Note that all of this is separate from the `#cgo` directive changes you made. (Those changes are correct.)

Reply:
Thanks for an explanation, now it makes perfect sense. Looks like the problem of both _native_ and _cross-compilcation_ can be resolved only by external tool. For example, shell script which will set `CGO_CGOCFLAGS` and `CGO_LDFLAGS` based on _target_ `GOOS` and `GOARCH`. But this is beyond the scope of my PR.